device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/services/Messages.java

```diff
@@ -24,21 +30,26 @@
 
     private final String dataPrefix = "data.";
 
-    private final DocumentClient docDbConnection;
-    private final String docDbCollectionLink;
+    private StorageType storageType;
+    private DocumentClient docDbConnection;
+    private String docDbCollectionLink;
+    private ITimeSeriesClient timeSeriesClient;
+
 
     @Inject
-    public Messages(
-        IServicesConfig servicesConfig) {
+    public Messages(IServicesConfig servicesConfig, ITimeSeriesClient timeSeriesClient) {
```

Comment:
We should create a IStorageClient interface and implement a concrete class that implements it for both DocDb and TSI storage. The DI should inject the needed concrete class and code in this file what is specific to CosmosDB should go in that new concrete class. messages.java should only need minimal changes.

Reply:
agree, this is ideal. but the two clients has a different set of methods. Alarms is also using the DocDB and require more interface methods like "upsertDocument", "deleteDocument" which are not equivalent in TSI.

Reply:
as discussed, we will refactor code here later.